modules/ob-example-configuration-for-defining-parameter-values.adoc

```diff
@@ -0,0 +1,165 @@
+
+[source,yaml]
+----
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: a-build
+  namespace: a-namespace
+spec:
+  paramValues:
+  - name: cache
+    value: disabled
+  strategy:
+    name: buildkit
+    kind: ClusterBuildStrategy
+  source:
+  # ...
+  output:
+  # ...
+----
+
+.Example: Creating a `ConfigMap` CR to control a parameter centrally
+If you want to use the `cache` parameter for multiple builds and control its usage centrally, then you can create a `ConfigMap` CR, as shown in the following example:
+
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: buildkit-configuration
+  namespace: a-namespace
+data:
+  cache: disabled
+----
+
+You can use the created `ConfigMap` CR as a parameter value in your `Build` CR, as shown in the following example:
+
+[source,yaml]
+----
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: a-build
+  namespace: a-namespace
+spec:
+  paramValues:
+  - name: cache
+    configMapValue:
+      name: buildkit-configuration
+      key: cache
+  strategy:
+    name: buildkit
+    kind: ClusterBuildStrategy
+  source:
+  # ...
+  output:
+  # ...
+----
+
+.Example: Defining a `build-args` parameter in a `Build` CR
+You can define a `build-args` parameter as an array. For example, if you use the `buildkit` strategy, you can define a `build-args` parameter to set the argument values in the container file. Values must be defined in the `KEY=VALUE` format as shown in the following example:
+
+[source,yaml]
+----
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: a-build
+  namespace: a-namespace
+spec:
+  paramValues:
+  - name: cache
+    configMapValue:
+      name: buildkit-configuration
+      key: cache
+  - name: build-args
+    values:
+    - value: NODE_VERSION=16
+  strategy:
+    name: buildkit
+    kind: ClusterBuildStrategy
+  source:
+  # ...
+  output:
+  # ...
+----
+
+.Example: Referencing a config map or secret in a `Build` CR
+You can also reference config maps or secrets in the `build-args` array parameter as shown in the following example:
+
+[source,yaml]
+----
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: a-build
+  namespace: a-namespace
+spec:
+  paramValues:
+  - name: cache
+    configMapValue:
+      name: buildkit-configuration
+      key: cache
+  - name: build-args
+    values:
+    - configMapValue:
+        name: project-configuration
+        key: node-version
+        format: NODE_VERSION=${CONFIGMAP_VALUE}
+    - value: DEBUG_MODE=true
+    - secretValue:
+        name: npm-registry-access
+        key: npm-auth-token
+        format: NPM_AUTH_TOKEN=${SECRET_VALUE}
+  strategy:
+    name: buildkit
+    kind: ClusterBuildStrategy
+  source:
+  # ...
+  output:
+  # ...
+----
+
+The above example sets the following values in the `build-args` array parameter:
```

Comment:
Is it possible to put these points in code callouts so its clearer which areas of the code they reference?